error: field 'arg1' has incomplete type .

<!-- Please make sure to run `v up` before reporting any issues as it may have already been fixed.
     It's also advisable to update all relevant modules using `v outdated` and `v install` -->

<!-- You can use `v doctor` to fill up the next fields -->
**V version:** 
**OS:** 
```
OS: linux, "Arch Linux"
Processor: 8 cpus, 64bit, little endian, Intel(R) Core(TM) i7-9700F CPU @ 3.00GHz
CC version: cc (GCC) 10.2.0

getwd: /storage/home/zero/Documents/programming/vemulator/assembler
vmodules: /home/zero/.vmodules
vroot: /usr/lib/vlang
vexe: /usr/lib/vlang/v
vexe mtime: 2021-04-13 21:20:37
is vroot writable: false
is vmodules writable: true
V full version: V 0.2.2 a2a18ef.8c95f07

Git version: git version 2.31.1
Git vroot status: weekly.2021.14-79-g8c95f075
.git/config present: true
thirdparty/tcc status: thirdparty-linux-amd64 7543de81-dirty
```
**What did you do?**

```v
module main 

fn broken() ([]string, ?IError) {
    return []string{}, error('')
}

fn main() {
    broken()
}
```

**What did you expect to see?**

Maybe an error, maybe not. Who knows!
 
**What did you see instead?**

```
==================
/tmp/v/assembler.13378355059222841325.tmp.c:881: error: field 'arg1' has incomplete type
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

If you were not working with C interop, this is a compiler bug, please raise an issue on GitHub:

https://github.com/vlang/v/issues/new/choose
```

--- a/vlib/v/checker/checker.v
+++ b/vlib/v/checker/checker.v
@@ -7458,6 +7458,18 @@ fn (mut c Checker) fn_decl(mut node ast.FnDecl) {
 				}
 			}
 		}
+		return_sym := c.table.get_type_symbol(node.return_type)
+		if return_sym.info is ast.MultiReturn {
+			for multi_type in return_sym.info.types {
+				if multi_type == ast.error_type {
+					c.error('type `IError` cannot be used in multi-return, return an option instead',
+						node.return_type_pos)
+				} else if multi_type.has_flag(.optional) {
+					c.error('option cannot be used in multi-return, return an option instead',
+						node.return_type_pos)
+				}
+			}
+		}
 	}
 	if node.return_type == ast.void_type {
 		for mut a in node.attrs {

--- a/vlib/v/checker/tests/ierror_in_return_tuple.out
+++ b/vlib/v/checker/tests/ierror_in_return_tuple.out
@@ -1,0 +1,12 @@
+vlib/v/checker/tests/ierror_in_return_tuple.vv:1:29: error: type `IError` cannot be used in multi-return, return an option instead
+    1 | fn return_ierror_in_tuple() (bool, IError) {
+      |                             ~~~~~~~~~~~~~~
+    2 |     return false, error('')
+    3 | }
+vlib/v/checker/tests/ierror_in_return_tuple.vv:5:29: error: option cannot be used in multi-return, return one option instead
+    3 | }
+    4 | 
+    5 | fn return_option_in_tuple() (bool, ?bool) {
+      |                             ~~~~~~~~~~~~~
+    6 |     return false, error('')
+    7 | }

--- a/vlib/v/checker/tests/ierror_in_return_tuple.out
+++ b/vlib/v/checker/tests/ierror_in_return_tuple.out
@@ -3,7 +3,7 @@ vlib/v/checker/tests/ierror_in_return_tuple.vv:1:29: error: type `IError` cannot
       |                             ~~~~~~~~~~~~~~
     2 |     return false, error('')
     3 | }
-vlib/v/checker/tests/ierror_in_return_tuple.vv:5:29: error: option cannot be used in multi-return, return one option instead
+vlib/v/checker/tests/ierror_in_return_tuple.vv:5:29: error: option cannot be used in multi-return, return an option instead
     3 | }
     4 | 
     5 | fn return_option_in_tuple() (bool, ?bool) {

--- a/vlib/v/checker/tests/ierror_in_return_tuple.vv
+++ b/vlib/v/checker/tests/ierror_in_return_tuple.vv
@@ -1,0 +1,7 @@
+fn return_ierror_in_tuple() (bool, IError) {
+	return false, error('')
+}
+
+fn return_option_in_tuple() (bool, ?bool) {
+	return false, error('')
+}

